File: justjavastaff/src/main/java/com/marknkamau/justjavastaff/ui/main/MainActivityPresenter.java
package com.marknkamau.justjavastaff.ui.main;

import android.content.SharedPreferences;

import com.marknkamau.justjavastaff.data.network.OrdersRepository;
import com.marknkamau.justjavastaff.models.Order;

import java.util.List;

class MainActivityPresenter {
    private final MainView view;
    private final SharedPreferences preferences;
    private OrdersRepository ordersRepository;

    MainActivityPresenter(MainView view, SharedPreferences preferences, OrdersRepository ordersRepository) {
        this.view = view;
        this.preferences = preferences;
        this.ordersRepository = ordersRepository;
    }

    void getOrders() {
        ordersRepository.getOrders(new OrdersRepository.OrdersListener() {
            @Override
            public void onSuccess(List<Order> orders) {
                if (orders.isEmpty()) {
                    view.displayNoOrders();
                } else {
                    view.displayAvailableOrders(orders);
                }
            }

            @Override
            public void onError(String reason) {
                view.displayMessage(reason);
            }
        });
    }
}

File: justjavastaff/src/main/java/com/marknkamau/justjavastaff/ui/main/MainActivityPresenter.kt
package com.marknkamau.justjavastaff.ui.main

import com.marknjunge.core.data.firebase.StaffDatabaseService
import com.marknjunge.core.model.Order
import com.marknjunge.core.model.OrderStatus
import com.marknkamau.justjavastaff.data.local.SettingsRespository

import timber.log.Timber

internal class MainActivityPresenter(private val view: MainView, private val settings: SettingsRespository, private val databaseService: StaffDatabaseService) {

    fun getOrders() {
        databaseService.getOrders(object : StaffDatabaseService.OrdersListener {
            override fun onSuccess(orders: List<Order>) {
                if (orders.isEmpty()) {
                    view.displayNoOrders()
                } else {
                    val statusSettings = settings.getStatusSettings()
                    val filteredOrders = mutableListOf<Order>()
                    orders.forEach {
                    Timber.d(it.status.toString())
                    Timber.d((it.status == OrderStatus.PENDING).toString())
                        if (it.status == OrderStatus.PENDING && statusSettings.pending)
                            filteredOrders.add(it)
                        if (it.status == OrderStatus.INPROGRESS && statusSettings.inProgress)
                            filteredOrders.add(it)
                        if (it.status == OrderStatus.COMPLETED && statusSettings.completed)
                            filteredOrders.add(it)
                        if (it.status == OrderStatus.DELIVERED && statusSettings.delivered)
                            filteredOrders.add(it)
                        if (it.status == OrderStatus.CANCELLED && statusSettings.cancelled)
                            filteredOrders.add(it)
                    }
                    view.displayAvailableOrders(filteredOrders)
                }
            }

            override fun onError(reason: String) {
                view.displayMessage(reason)
            }

        })

    }
}
